modules/heat_conduction/src/materials/SideSetHeatTransferMaterial.C

```diff
@@ -0,0 +1,125 @@
+                                "Neighbor face reflectivity, uses (1-emissivity) if not provided.");
+  return params;
+}
+
+SideSetHeatTransferMaterial::SideSetHeatTransferMaterial(const InputParameters & parameters)
+  : InterfaceMaterial(parameters),
+    _kgap(isParamValid("conductivity_temperature_function")
+              ? getFunction("conductivity_temperature_function")
+              : getFunction("conductivity")),
+    _Tk(isCoupled("gap_temperature") ? &coupledValue("gap_temperature") : nullptr),
+    _dgap(getFunction("gap_length")),
+    _Tb(getFunction("Tbulk")),
+    _hp(getFunction("h_master")),
+    _hm(getFunction("h_neighbor")),
+    _eps_p(getFunction("emissivity_master")),
+    _eps_m(getFunction("emissivity_neighbor")),
+    _rho_p(isParamValid("reflectivity_master") ? &getFunction("reflectivity_master") : nullptr),
+    _rho_m(isParamValid("reflectivity_neighbor") ? &getFunction("reflectivity_neighbor") : nullptr),
+    _cond(declareProperty<Real>("gap_conductance")),
+    _Tbulk(declareProperty<Real>("gap_Tbulk")),
+    _h_master(declareProperty<Real>("gap_h_master")),
+    _h_neighbor(declareProperty<Real>("gap_h_neighbor")),
+    _emmissivity_eff_master(declareProperty<Real>("gap_emissivity_eff_master")),
+    _emmissivity_eff_neighbor(declareProperty<Real>("gap_emissivity_eff_neighbor")),
+    _sigma(SIGMA)
+{
+  if ((parameters.isParamSetByUser("conductivity") ||
+       isParamValid("conductivity_temperature_function")) &&
+      !parameters.isParamSetByUser("gap_length"))
+    mooseError("gap_length not set, but conduction term requested.");
+  if (parameters.isParamSetByUser("gap_length") &&
+      !(parameters.isParamSetByUser("conductivity") ||
+        isParamValid("conductivity_temperature_function")))
+    mooseError("conductivity not set, but conduction term requested.");
+  if (isParamValid("conductivity_temperature_function") &&
+      parameters.isParamSetByUser("conductivity"))
+    mooseError("Cannot specify both conductivity and conductivity_temperature_function.");
+  if (isParamValid("conductivity_temperature_function") && !_Tk)
+    mooseError("Variable specification for temp needed if specifying a temperature dependent "
+               "conductivity.");
+
+  if (parameters.isParamSetByUser("h_master") && !parameters.isParamSetByUser("h_neighbor"))
+    mooseError("h_neighbor not set, but convection term requested.");
+  if (parameters.isParamSetByUser("h_neighbor") && !parameters.isParamSetByUser("h_master"))
+    mooseError("h_master not set, but convection term requested.");
+
+  if (parameters.isParamSetByUser("emissivity_master") &&
+      !parameters.isParamSetByUser("emissivity_neighbor"))
+    mooseError("emissivity_neighbor not set, but radiation term requested");
+  if (parameters.isParamSetByUser("emissivity_neighbor") &&
+      !parameters.isParamSetByUser("emissivity_master"))
+    mooseError("emissivity_master not set, but radiation term requested");
```

Comment:
Looks like you could switch nearly all of these to param errors. You can just pick one when there are multiple parameters involved. What happens is you get the line number of one of the affected parameters.